src/Form/CommerceBundleEntityDeleteFormBase.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Drupal\commerce\Form;
+
+use Drupal\Core\Entity\EntityDeleteForm;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Builds the form to delete an order item type.
+ */
+class CommerceBundleEntityDeleteFormBase extends EntityDeleteForm {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    /** @var \Drupal\Core\Config\Entity\ConfigEntityType $bundle_entity_type */
+    $bundle_entity_type = $this->entityTypeManager->getDefinition($this->entity->getEntityTypeId());
+    /** @var \Drupal\Core\Entity\ContentEntityType $content_entity_type */
+    $content_entity_type = $this->entityTypeManager->getDefinition($bundle_entity_type->getBundleOf());
+    $usage_count = $this->entityTypeManager->getStorage($content_entity_type->id())
+      ->getQuery()
+      ->condition('type', $this->entity->id())
```

Comment:
You need to check the content entity type's definition to get the proper bundle key.